### ISSUE TITLE
Minor fixes to the dump PR

### ISSUE THREAD

**Issue by shankari** [MobilityNet/mobilitynet.github.io#23]
Minor fixes to:
https://github.com/MobilityNet/mobilitynet-analysis-scripts/pull/54

- Add a comment explaining the reason for `sys.maxsize`
- rename `run_full_pipeline`
- in `trajectory_evaluation.ipynb`, the ground truth retrieval can use `SpecDetails`, doesn't need to use `FileSpecDetails`
- Ditto for `trajectory_evaluation_spatio_temporal.ipynb`
- And `trajectory_evaluation_timestamp_metric_selection.ipynb`

**Comment by singhish** [comment]
One thing to note regarding the last 3 points -- in order for the notebooks to run, it is necessary for `SpecDetails.get_concat_trajectories`, `SpecDetails.get_shapes_for_leg`, and `SpecDetails.get_geojson_for_leg` to be fixed to accommodate the updated `start_loc`/`end_loc`/`route_coords`/`loc` structures. They are currently handling these keys as dicts when they are now lists, which affects the logic that is currently in the notebooks. I was working on a fix today to address this but decided to scrap it as would probably make more sense to talk this over with you tomorrow @shankari.

**Comment by shankari** [comment]
@singhish yes you should change all the `get_*` methods to work with the updated structures.
By scrap, I assume you didn't actually throw out the code!

**Comment by singhish** [comment]
The `trajectory_evaluation.ipynb` notebook is now fully working. There are a few things I ran into with the other notebooks:
- `trajectory_evaluation_spatio_temporal.ipynb`: the `spatial_algo_df` is for some reason empty, and has nothing to do with `get_ground_truth_for_leg`. Blocked here, will touch base with you Monday @shankari.
- `trajectory_evaluation_timestamp_metric_selection.ipynb`: this notebook uses a spec that doesn't have reroutes (`sd_la`), need to make a design decision regarding `get_ground_truth_for_leg` to make `start_ts` and `end_ts` optional arguments in this case.

**Comment by shankari** [comment]
> trajectory_evaluation_timestamp_metric_selection.ipynb: this notebook uses a spec that doesn't have reroutes (sd_la), need to make a design decision regarding get_ground_truth_for_leg to make start_ts and end_ts optional arguments in this case.

@singhish Use the start and end timestamps of the entire spec. Just like we did for spec creation.
